Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm

```diff
@@ -243,21 +248,37 @@ bool isParentProcessAFullWebBrowser(AuxiliaryProcess& auxiliaryProcess)
     return fullWebBrowser || isRunningTest(WebCore::applicationBundleIdentifier());
 }
 
-static bool isFullWebBrowser(const String& bundleIdentifier)
+bool isFullWebBrowserOrRunningTest(const String& bundleIdentifier)
 {
     ASSERT(!isInWebKitChildProcess());
 
+#if ENABLE(APP_BOUND_DOMAINS)
     static bool fullWebBrowser = WTF::processHasEntitlement("com.apple.developer.web-browser"_s);
+#elif PLATFORM(MAC)
+    static bool fullWebBrowser;
+    static std::once_flag once;
+    std::call_once(once, [] {
+        NSURL *currentURL = [[NSBundle mainBundle] bundleURL];
+        NSArray<NSURL *> *httpURLs = [[NSWorkspace sharedWorkspace] URLsForApplicationsToOpenURL:[NSURL URLWithString:@"http:"]];
+        bool canOpenHTTP = [httpURLs containsObject:currentURL];
+        NSArray<NSURL *> *httpsURLs = [[NSWorkspace sharedWorkspace] URLsForApplicationsToOpenURL:[NSURL URLWithString:@"https:"]];
+        bool canOpenHTTPS = [httpsURLs containsObject:currentURL];
+        fullWebBrowser = canOpenHTTPS && canOpenHTTP;
+    });
```

Comment:
Will for example MiniBrowser be classified as a full Web browser with this implementation? Will the return value be different in the UI process and the WebContent process?

Reply:
> Will for example MiniBrowser be classified as a full Web browser with this implementation?

No, because MiniBrowser does not register for these two schemes. This is the heuristics used for identifying browsers by system, see https://developer.apple.com/documentation/xcode/preparing-your-app-to-be-the-default-browser-or-email-client, so I think this is a bug in MiniBrowser.

> Will the return value be different in the UI process and the WebContent process?

This function is not supposed to be called in web process and we have assertion above:
`ASSERT(!isInWebKitChildProcess());`
Web process should use `isParentProcessAFullWebBrowser`.